dev/TabView/TabViewItemAutomationPeer.cpp

```diff
@@ -24,3 +24,25 @@ winrt::AutomationControlType TabViewItemAutomationPeer::GetAutomationControlType
     return winrt::AutomationControlType::TabItem;
 }
 
+
+winrt::hstring TabViewItemAutomationPeer::GetNameCore()
+{
+    winrt::hstring returnHString = __super::GetNameCore();
+
+    // If a name hasn't been provided by AutomationProperties.Name in markup:
+    if (returnHString.empty())
+    {
+        if (auto tvi = Owner().try_as<winrt::TabViewItem>())
+        {
+            if (auto content = tvi.Header())
+            {
+                if (auto stringableName = content.try_as<winrt::IStringable>())
```

Comment:
I see similar but not identical code in TreeViewItemAutomationPeer -- can you factor that out as a helper to turn an object into an accessible name?

https://github.com/microsoft/microsoft-ui-xaml/blob/fcaff88fbe7db256e7092bfdff650aa506cd65eb/dev/TreeView/TreeViewItemAutomationPeer.cpp#L92

Reply:
Maybe....

The problem is that the helper method would need to be a bit on the complex side:
1. It needs to call __super::GetNameCore(). I'm not actually sure what is the correct way to do this from a non-member function. Even if the actual type of __super is passed as a template parameter, I don't know how to call it. 
2. It needs to cast to a type and call a property on the item (e.g. TabViewItem.Header()), the type could be a template param, but then the property would be... a lambda?

Adding a helper function might not be worth the extra complexity, but I'll have a look.

Reply:
I have created a helper function to convert an IInspectable to a hstring. I didn't try to abstract the rest of the structure of the GetNameCore function, since it adds more complexity than appropriate.